server/libs/domain/src/storage-template/storage-template.core.ts

```diff
@@ -33,12 +34,14 @@ export class StorageTemplateCore {
     this.configCore.config$.subscribe((config) => this.onConfig(config));
   }
 
-  public async getTemplatePath(asset: AssetEntity, filename: string): Promise<string> {
+  public async getTemplatePath(asset: AssetEntity, metadata: MoveAssetMetadata): Promise<string> {
+    const { username, filename } = metadata;
+
     try {
       const source = asset.originalPath;
       const ext = path.extname(source).split('.').pop() as string;
       const sanitized = sanitize(path.basename(filename, `.${ext}`));
-      const rootPath = this.storageCore.getFolderLocation(StorageFolder.LIBRARY, asset.ownerId);
+      const rootPath = this.storageCore.getFolderLocation(StorageFolder.LIBRARY, sanitize(username || asset.ownerId));
```

Comment:
The sanitized username can be empty (username `..` for example) and then no subfolder would be created. Multiple users can also end up using the same folder by abusing the sanitation: `user` and `user..` for example

Reply:
You could add `@Transform(({ value }) => sanitize(value))` to the DTO. The validators should prevent empty usernames and the unique constraint takes care of duplicate usernames

Reply:
Good idea, I'll give that a shot.